### PR TITLE
test: [MSSQL] data migration from legacy instance

### PR DESCRIPTION
**Pull request by zucchinidev**
### Checklist:

* [ ] Have you added Release Notes in the docs repositories?
* [x] Have you ran `make run-integration-tests` and `make run-terraform-tests`?
* [x] Have you ran acceptance tests for the service under change?
* [x] Have you followed the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)?

### CODE REVIEW

File: acceptance-tests/mssql_data_migration_test.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+		By("reading value from legacy service instance")
+		got := golangAppOne.GET("%s/%s?tls=disable", schema, key).String()
+		Expect(got).To(Equal(value))
+
+		By("waiting for the replication instance to be ready")
+		replicationInstance.Wait()
+
+		legacyServiceInstanceUUID := legacyServiceInstance.GUID()
+		password := getPassword(
```
Comment (pivotal-marcela-campo) [inline acceptance-tests/mssql_data_migration_test.go:78]:
```suggestion
		sourceAdminPassword := getPassword(
```

File: acceptance-tests/mssql_data_migration_test.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+			"/bin/bash", "-c",
+			fmt.Sprintf(
+				`/usr/local/bin/legacy_data_assister --instance-uuid="%s"`,
+				legacyServiceInstanceUUID,
+			),
+		)
+
+		Expect(password).NotTo(BeEmpty())
```
Comment (pivotal-marcela-campo) [inline acceptance-tests/mssql_data_migration_test.go:86]:
```suggestion
		Expect(sourceAdminPassword).NotTo(BeEmpty())
```

File: acceptance-tests/mssql_data_migration_test.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+			Username string `mapstructure:"username"`
+			Password string `mapstructure:"password"`
+			Port     int    `mapstructure:"port"`
+		}
+		Expect(mapstructure.Decode(sourceCreds, &sourceReceiver)).NotTo(HaveOccurred())
+		sourceEndpoint := dms.CreateEndpoint(dms.CreateEndpointParams{
+			EndpointType:    dms.Source,
+			EnvironmentName: metadata.Name,
+			Username:        masterUsername,
+			Password:        password,
```
Comment (pivotal-marcela-campo) [inline acceptance-tests/mssql_data_migration_test.go:102]:
```suggestion
			Password:        sourceAdminPassword,
```

File: acceptance-tests/mssql_data_migration_test.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+	envMasterUsername = "MasterUsername"
+)
+
+// The legacy_data_assister binary must be installed. See script in CI project and instructions on the Wiki
+var _ = Describe("MSSQL data migration", Label("mssql-migration"), func() {
+	var masterUsername string
+
+	It("can migrate data from the previous broker to the CSB", func() {
+		By("reading legacy broker RDS data")
+		masterUsername = os.Getenv(envMasterUsername)
```
Comment (pivotal-marcela-campo) [inline acceptance-tests/mssql_data_migration_test.go:31]:
```suggestion
		sourceAdminUsername = os.Getenv(envMasterUsername)
```

File: acceptance-tests/mssql_data_migration_test.go [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+		Expect(masterUsername).NotTo(BeEmpty(), "The MasterUsername environment variable is mandatory")
```
Comment (pivotal-marcela-campo) [inline acceptance-tests/mssql_data_migration_test.go:32]:
```suggestion
		Expect(sourceAdminUsername).NotTo(BeEmpty(), "The MasterUsername environment variable is mandatory")
```

File: acceptance-tests/mssql_data_migration_test.go [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+			Server   string `mapstructure:"hostname"`
+			Username string `mapstructure:"username"`
+			Password string `mapstructure:"password"`
+			Port     int    `mapstructure:"port"`
+		}
+		Expect(mapstructure.Decode(sourceCreds, &sourceReceiver)).NotTo(HaveOccurred())
+		sourceEndpoint := dms.CreateEndpoint(dms.CreateEndpointParams{
+			EndpointType:    dms.Source,
+			EnvironmentName: metadata.Name,
+			Username:        masterUsername,
```
Comment (pivotal-marcela-campo) [inline acceptance-tests/mssql_data_migration_test.go:101]:
```suggestion
			Username:        sourceAdminUsername,
```

File: acceptance-tests/mssql_data_migration_test.go [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+
+		Expect(golangAppOne.GET("%s/%s?tls=disable", schema, key).String()).To(Equal(value))
+	})
+})
+
+func getPassword(cmd string, args ...string) string {
+	command := exec.Command(cmd, args...)
+	session, err := gexec.Start(command, GinkgoWriter, GinkgoWriter)
+	Expect(err).NotTo(HaveOccurred())
+	Eventually(session, time.Minute).Should(gexec.Exit(0))
```
Comment (pivotal-marcela-campo) [inline acceptance-tests/mssql_data_migration_test.go:167]: do we want to register this as a helper?

**Review by pivotal-marcela-campo — CHANGES_REQUESTED** [review]

File: acceptance-tests/mssql_data_migration_test.go (hunk 7 above)
Comment (zucchinidev) [reply acceptance-tests/mssql_data_migration_test.go:167]: great point! Yes. I will modify it now

**Review by pivotal-marcela-campo — APPROVED** [review]

**Pull request merged**